gpio/__init__.py

```diff
@@ -0,0 +1,280 @@
+# -*- coding: utf-8 -*-
+__version__ = '1.0.0'
+
+from threading import Lock
+import os
+
+
+_export_lock = Lock()
+_open_pins = {}
+
+
+GPIO_ROOT = '/sys/class/gpio'
+GPIO_EXPORT = os.path.join(GPIO_ROOT, 'export')
+GPIO_UNEXPORT = os.path.join(GPIO_ROOT, 'unexport')
+FMODE = 'w+'  # w+ overwrites and truncates existing files
+IN, OUT = 'in', 'out'
+LOW, HIGH = 0, 1
+BCM = None  # Exists for RPi.GPIO compatibility
+
+
+class GPIOPin(object):

```

Comment:
I think it makes sense to use these classes for the existing functions. "register" will just be putting one of these instances in a map.

BTW, the terminology for these kind of instances (where there can only be one instance per pin) is a `singleton`. Worth documenting that and asserting that invariant in the `__init__`.